src/cli.zig

```diff
@@ -181,44 +210,68 @@ pub const OptionAlias = enum(u7) {
+};
+
+/// Returns a struct with all the same fields as input struct `S`, but where each
+/// field is of type `OptionInfo(FieldType)`.
+///
+/// Certain groups of options may appear together repeatedly across subcommands;
+/// for this reason, struct fields in the struct passed to `CommandInfo` are described
+/// using this type, allowing re-use of those struct types across subcommands.
+///
+/// Each of the option infos in the struct are embedded into the parent pool of
+/// options; an error is issued if any name or alias collisions occur.
+pub fn OptionInfoGroup(comptime S: type) type {
+    const Type = std.builtin.Type;
+    const s_info = @typeInfo(S).Struct;
+
+    var sub_fields: [s_info.fields.len]Type.StructField = undefined;
+    for (&sub_fields, s_info.fields) |*new_s_field, s_field| {
+        if (@typeInfo(s_field.type) == .Union or
+            (@typeInfo(s_field.type) == .Optional and
+            @typeInfo(@typeInfo(s_field.type).Optional.child) == .Union))
+        {
+            @compileError("The subcommand field cannot be part of an option group");
+        }
 
-    pub fn from(char: u8) OptionAlias {
-        return switch (char) {
-            'A'...'Z',
-            'a'...('h' - 1),
-            ('h' + 1)...'z',
-            => |c| @enumFromInt(c),
-            else => .none,
+        new_s_field.* = .{
+            .name = s_field.name,
+            .type = OptionInfo(s_field.type),
+            .default_value = null,
+            .is_comptime = false,
+            .alignment = 0,
         };
     }
-};
+
+    return @Type(.{ .Struct = .{
+        .layout = .auto,
+        .backing_integer = null,
+        .fields = &sub_fields,
+        .is_tuple = false,
+        .decls = &.{},
+    } });
+}
 
 pub const ParseCmdError = error{
     /// Unrecognized `--{name}`.
     UnrecognizedOptionName,
     /// Unrecognized `-{c}`.
     UnrecognizedOptionAlias,
-
     /// Unexpected positional in command with no subcommand and not expecting positionals.
     UnexpectedPositional,
-
+    /// Unrecognized command string.
     UnrecognizedCommand,
-    MissingOptions,
-    MissingCommand,
 } || ParseOptionKeyMaybeValStrError ||
     ParseSingleOptValueError ||
     std.mem.Allocator.Error ||
     std.os.windows.SetConsoleTextAttributeError; // for the TTY for the help message
 
 pub fn Parser(
-    /// Must be a struct type containing zero or more option fields, and zero or one
-    /// sub-command fields which is a tagged union (possibly optional) wherein each
+    /// Must be a struct type containing zero or more option fields, and optionally
+    /// one sub-command field which is an optional tagged union, wherein each
     /// member is a sub-command struct following the same description as this one,
     /// recursively.
-    ///
-    /// An option field must either be a parsed value such as a
-    /// boolean, integer, enum, string, or list or optional there-of,
-    /// or a struct of parsed values.
+    /// See `CommandInfo`.
```

Comment:
is this supposed to be one line lower?

Reply:
No, it's in correct position, it's making reference to the doc comment of `CommandInfo`, which is relevant here.